### PR TITLE
Email 2fa through django_otp email plugin

### PR DESCRIPTION
**Pull request by claudep**
This is a follow-up pull request after #475, but based on the `EmailDevice` class from `django_otp`.

It also includes the commits from #476 and #477 (may be rebased when those are committed).

### CODE REVIEW

File: two_factor/plugins/email/forms.py
Comment (claudep) [inline two_factor/plugins/email/forms.py:28]: We might remove those field overrides if #479 is merged, to be tested.

**Comment by dchukhin** [comment]
Thanks for this work @claudep!
I’m not a maintainer of this library, but I tried your code out locally, and it was very easy to set up 2FA by email ✔️ .
 I did notice an issue for the following scenario, so [I made a pull request against your fork](https://github.com/claudep/django-two-factor-auth/pull/2) to fix it:
1. User without 2FA logs in
2. User sets up 2FA by email
3. User logs out
4. User goes to log back in, enters username and password
5. User is not required to enter 2FA code

File: two_factor/plugins/email/forms.py
Comment (moggers87) [reply two_factor/plugins/email/forms.py:28]: #479 has been merged

**Comment by claudep** [comment]
@moggers87 I think this is now ready for review.

File: two_factor/apps.py
Comment (moggers87) [reply two_factor/apps.py:17]: Just to be clear: this one is a comment and not something I'm requesting be changed.

**Review by moggers87 — CHANGES_REQUESTED** [review]

File: two_factor/apps.py
Comment (claudep) [reply two_factor/apps.py:17]: You are absolutely right, we should follow the same logic as for other plugins. Thanks for catching!

**Comment by claudep** [comment]
I still have 2-3 things to fix, hopefully for tomorrow.

File: tests/test_email.py
Comment (claudep) [reply tests/test_email.py]: I removed those lines, they are not absolutely needed.

File: two_factor/apps.py
Comment (claudep) [reply two_factor/apps.py]: Updated to follow the other plugins logic.

**Review by moggers87 — APPROVED** [review]
LGTM, feel free to merge if there were no other changes :+1:

**Pull request merged**
